### PR TITLE
fix: docker-publish.yml cosign command line args when multiple tags are present

### PR DESCRIPTION
**Pull request by jamesmoore**
With scheduled builds the output docker image has two tags - branch name (eg, main) and nightly.

The current template is passing these as a single argument into the cosign sign command. This is causing failures of the workflow when run with a scheduled trigger.

This change splits them using xargs and runs multiple cosign commands, one for each tag.

Tested by running against one of my own repos, eg: https://github.com/jamesmoore/mp3info/runs/7087514897?check_suite_focus=true#step:8:2

### CODE REVIEW

**Review by arnuphong — APPROVED** [review]

File: ci/docker-publish.yml [hunk 1]
```diff
@@ -90,4 +90,4 @@ jobs:
           COSIGN_EXPERIMENTAL: "true"
         # This step uses the identity token to provision an ephemeral certificate
         # against the sigstore community Fulcio instance.
-        run: cosign sign ${{ steps.meta.outputs.tags }}@${{ steps.build-and-push.outputs.digest }}
+        run: echo "${{ steps.meta.outputs.tags }}" | xargs -I {} cosign sign {}@${{ steps.build-and-push.outputs.digest }}
```
Comment (arnuphong) [inline ci/docker-publish.yml:93]: #1622

Comment (arnuphong) [reply ci/docker-publish.yml:93]: #1636

Comment (arnuphong) [reply ci/docker-publish.yml:93]: #202

Comment (arnuphong) [reply ci/docker-publish.yml:93]: #439

File: ci/docker-publish.yml [hunk 2]
```diff
@@ -90,4 +90,4 @@ jobs:
           COSIGN_EXPERIMENTAL: "true"
         # This step uses the identity token to provision an ephemeral certificate
         # against the sigstore community Fulcio instance.
-        run: cosign sign ${{ steps.meta.outputs.tags }}@${{ steps.build-and-push.outputs.digest }}
```
Comment (arnuphong) [inline ci/docker-publish.yml:93]: @jamesmoore

Comment (arnuphong) [inline ci/docker-publish.yml:93]: @josh

Comment (arnuphong) [inline ci/docker-publish.yml:93]: @arnuphong

File: ci/docker-publish.yml [hunk 3]
```diff
@@ -90,4 +90,4 @@ jobs:
           COSIGN_EXPERIMENTAL: "true"
```
Comment (arnuphong) [inline ci/docker-publish.yml:90]: .

**Review by arnuphong — CHANGES_REQUESTED** [review]

File: ci/docker-publish.yml [hunk 4]
```diff
@@ -90,4 +90,4 @@ jobs:
...
         # This step uses the identity token to provision an ephemeral certificate
         # against the sigstore community Fulcio instance.
```
Comment (arnuphong) [inline ci/docker-publish.yml:92]: [#1306](https://github.com/actions/starter-workflows/issues/1306)

**Review by arnuphong — APPROVED** [review]

**Review by arnuphong — COMMENTED** [review]
#

**Review by CJCShadowsan — APPROVED** [review]
works for me.

**Review by Phantsure — APPROVED** [review]
This seems to fix an issue. Confirmed my different people

**Pull request merged**